src/+io/prepareInputData.m

```diff
@@ -19,16 +19,17 @@
 
     %% Explicitly load all variables into the workspace
     TimeProperties = load(forcing_global_path, 'DELT', 'Dur_tot');
-    SiteProperties = load(forcing_global_path, 'IGBP_veg_long', 'latitude', 'longitude', 'reference_height', 'canopy_height', 'sitename');
+    SiteProperties = load(forcing_global_path, 'latitude', 'longitude', 'reference_height', 'canopy_height', 'sitename');
 
     SoilProperties = load(soil_global_path, 'SaturatedK', 'SaturatedMC', 'ResidualMC', 'Coefficient_n', 'Coefficient_Alpha', ...
                           'porosity', 'FOC', 'FOS', 'MSOC', 'Coef_Lamda', 'fieldMC', 'fmax', 'theta_s0', 'Ks0');
 
     % Convert the int vectors back to strings
     SiteProperties.sitename = char(SiteProperties.sitename);
-    SiteProperties.IGBP_veg_long = char(SiteProperties.IGBP_veg_long);
 
-    % The model expects the char vector to be transposed:
-    SiteProperties.IGBP_veg_long = transpose(SiteProperties.IGBP_veg_long);
+    % Convert the 1-D int vector into a vector of strings;
+    load(forcing_global_path, 'IGBP_veg_long');
+    nRows = size(IGBP_veg_long)(2) / TimeProperties.Dur_tot;
+    SiteProperties.landcoverClass = cellstr(char(reshape(IGBP_veg_long, [nRows, TimeProperties.Dur_tot])'));
```

Comment:
Can this part `reshape(IGBP_veg_long, [nRows, TimeProperties.Dur_tot])'` be done by `pystemmusscope`? 
Also, can you comment on why `IGBP_veg_long` is stored as `int`?

Reply:
In PyStemmusScope we use hdf5storage to write the `.mat` file. 

In this process, the string array is converted to a 1-D vector of integers (or at least, that is how Matlab/Octave's `load` interprets it).

I don't see any clear way in [hdf5storage.savemat](https://pythonhosted.org/hdf5storage/hdf5storage.html#hdf5storage.savemat) to avoid this.

In general Matlab is quite wonky with chars/strings/matrices of chars and cellstrings. a `cellstr` is the most similar to a python list or numpy array of strings.